codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/s3/S3ExpressIntegration.kt

```diff
@@ -0,0 +1,75 @@
+package software.amazon.smithy.aws.swift.codegen.customization.s3
+
+import software.amazon.smithy.aws.swift.codegen.swiftmodules.AWSSDKIdentityTypes
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftDelegator
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.config.ClientConfiguration
+import software.amazon.smithy.swift.codegen.config.ConfigProperty
+import software.amazon.smithy.swift.codegen.config.DefaultProvider
+import software.amazon.smithy.swift.codegen.core.SwiftCodegenContext
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.toGeneric
+import software.amazon.smithy.swift.codegen.swiftmodules.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.swiftmodules.SwiftTypes
+
+class S3ExpressIntegration : SwiftIntegration {
+
+    override fun enabledForService(
+        model: Model,
+        settings: SwiftSettings,
+    ): Boolean = model.expectShape<ServiceShape>(settings.service).isS3
+
+    override fun writeAdditionalFiles(
+        ctx: SwiftCodegenContext,
+        protocolGenerationContext: ProtocolGenerator.GenerationContext,
+        delegator: SwiftDelegator
+    ) {
```

Comment:
This method generates a file that creates a concrete implementation of the `S3ExpressCreateSessionClient` protocol.

An instance of this class is passed into the S3 credential provider, and allows that credential provider to call S3 without actually depending on S3.

Reply:
See the generated file above:
https://github.com/awslabs/aws-sdk-swift/pull/1906/files/fec058fa0f5518470dcf1479b5b38365db683302#diff-9eaf9732200d9284b72d8a77b8d86ee2076509f03417f0ba54ad5434b3258d15